src/handlers/payout/action.ts

```diff
@@ -1,21 +1,82 @@
-import { getWalletAddress, getWalletMultiplier } from "../../adapters/supabase";
+import { BigNumber, ethers } from "ethers";
+import { getPenalty, getWalletAddress, getWalletMultiplier, removePenalty } from "../../adapters/supabase";
 import { getBotConfig, getBotContext, getLogger } from "../../bindings";
-import { addLabelToIssue, deleteLabel, generatePermit2Signature, getAllIssueComments, getTokenSymbol } from "../../helpers";
-import {UserType, Payload, StateReason } from "../../types";
+import {
+  addLabelToIssue,
+  deleteLabel,
+  generatePermit2Signature,
+  getAllIssueAssignEvents,
+  getAllIssueComments,
+  getTokenSymbol,
+  wasIssueReopened,
+} from "../../helpers";
+import { UserType, Payload, StateReason } from "../../types";
 import { shortenEthAddress } from "../../utils";
 import { bountyInfo } from "../wildcard";
 
 export const handleIssueClosed = async () => {
   const context = getBotContext();
   const {
-    payout: { paymentToken, rpc },
+    payout: { paymentToken, rpc, permitBaseUrl, chainId },
     mode: { autoPayMode },
   } = getBotConfig();
   const logger = getLogger();
   const payload = context.payload as Payload;
   const issue = payload.issue;
   if (!issue) return;
 
+  const comments = await getAllIssueComments(issue.number);
+
+  const wasReopened = await wasIssueReopened(issue.number);
+  const claimUrlRegex = new RegExp(`\\((${permitBaseUrl}\\?claim=\\S+)\\)`);
```

Comment:
Should this regex include the new `?network=100` query parameter?

Reply:
Old permit urls don't have network parameter yet and this regex is primarily meant just for finding the permit url, then later the network is extracted if it's available

Reply:
On one hand, I'm not sure that we need backward compatibility. On the other hand, as we contribute to modify the URL parameters for the cashouts, this seems like an unnecessarily brittle approach. 

Perhaps we need a better strategy for permit URL detection. Definitely make sure to check that the comment is issued by `@UbiquiBot` and perhaps we should focus on finding `pay.ubq.fi` and an anchor?

@rndquu rfc

Reply:
https://github.com/ubiquity/ubiquibot/issues/526